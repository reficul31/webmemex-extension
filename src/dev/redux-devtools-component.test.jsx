/* eslint-env jest */

import React from 'react'
import DevTools from 'src/dev/redux-devtools-component'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import overview from 'src/overview'
import { createStore, compose, combineReducers } from 'redux'

const reducer = combineReducers({
    overview: overview.reducer,
})

const enhancers = [DevTools.instrument()]
const enhancer = compose(...enhancers)

describe('DevTools', () => {
    test('should test against the given snapshot', () => {
        const store = createStore(
            reducer,
            undefined,
            enhancer,
        )
        const tree = renderer.create(
            <Provider store={store}>
                <DevTools />
            </Provider>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
