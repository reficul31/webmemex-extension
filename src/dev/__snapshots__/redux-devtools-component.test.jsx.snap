// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevTools should test against the given snapshot 1`] = `
<div
  style={
    Object {
      "height": 0,
      "left": 0,
      "position": "fixed",
      "top": 0,
      "width": 0,
      "zIndex": 99999999,
    }
  }
>
  <div
    style={
      Object {
        "MozBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "MozTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out,opacity 0.01s linear 0.2s",
        "OBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "OTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out,opacity 0.01s linear 0.2s",
        "WebkitBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "WebkitTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out,opacity 0.01s linear 0.2s",
        "background": "white",
        "boxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "height": "100%",
        "left": 1024,
        "msBoxShadow": "0 0 4px rgba(0, 0, 0, 0.3)",
        "msTransition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out,opacity 0.01s linear 0.2s",
        "opacity": 0,
        "position": "fixed",
        "top": 0,
        "transition": "left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out,opacity 0.01s linear 0.2s",
        "width": "30%",
        "zIndex": 1,
      }
    }
  >
    <div
      onMouseDown={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "left": -5,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "width": 10,
          "zIndex": 2,
        }
      }
    />
    <div
      style={
        Object {
          "height": "100%",
          "overflow": "auto",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#2A2F3A",
            "direction": "ltr",
            "fontFamily": "monaco, Consolas, Lucida Console, monospace",
            "height": "100%",
            "minWidth": 300,
            "overflowY": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderColor": "#4F5A65",
              "display": "flex",
              "flexDirection": "row",
              "textAlign": "center",
              "zIndex": 1,
            }
          }
        >
          <a
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "backgroundColor": "#4F5A65",
                "borderRadius": 3,
                "color": "white",
                "cursor": "pointer",
                "display": "inline-block",
                "flexGrow": 1,
                "fontSize": "0.8em",
                "fontWeight": "bold",
                "marginBottom": 5,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 5,
                "padding": 4,
                "textDecoration": "none",
              }
            }
          >
            Reset
          </a>
          <a
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 3,
                "color": "white",
                "cursor": "text",
                "display": "inline-block",
                "flexGrow": 1,
                "fontSize": "0.8em",
                "fontWeight": "bold",
                "marginBottom": 5,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 5,
                "opacity": 0.2,
                "padding": 4,
                "textDecoration": "none",
              }
            }
          >
            Revert
          </a>
          <a
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 3,
                "color": "white",
                "cursor": "text",
                "display": "inline-block",
                "flexGrow": 1,
                "fontSize": "0.8em",
                "fontWeight": "bold",
                "marginBottom": 5,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 5,
                "opacity": 0.2,
                "padding": 4,
                "textDecoration": "none",
              }
            }
          >
            Sweep
          </a>
          <a
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 3,
                "color": "white",
                "cursor": "text",
                "display": "inline-block",
                "flexGrow": 1,
                "fontSize": "0.8em",
                "fontWeight": "bold",
                "marginBottom": 5,
                "marginLeft": 3,
                "marginRight": 3,
                "marginTop": 5,
                "opacity": 0.2,
                "padding": 4,
                "textDecoration": "none",
              }
            }
          >
            Commit
          </a>
        </div>
        <div
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "absolute",
              "right": 0,
              "top": 30,
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "color": "#FFFFFF",
                  "opacity": 1,
                  "textDecoration": "none",
                }
              }
            >
              <div
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "backgroundColor": "#4F5A65",
                    "color": "#FFFFFF",
                    "cursor": "default",
                    "display": "block",
                    "opacity": 1,
                  }
                }
              >
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "paddingBottom": 7,
                      "paddingLeft": 16,
                      "paddingTop": 8,
                    }
                  }
                >
                  @@INIT
                </div>
                <div
                  style={
                    Object {
                      "backgroundColor": "#2A2F3A",
                      "margin": 0,
                      "overflow": "auto",
                      "paddingLeft": 16,
                    }
                  }
                >
                  
                </div>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 16,
                  }
                }
              >
                <ul
                  style={
                    Object {
                      "MozUserSelect": "none",
                      "WebkitUserSelect": "none",
                      "backgroundColor": "#2A2F3A",
                      "border": 0,
                      "listStyle": "none",
                      "marginBottom": "0.5em",
                      "marginLeft": "0.125em",
                      "marginRight": 0,
                      "marginTop": "0.5em",
                      "padding": 0,
                    }
                  }
                >
                  <li
                    style={
                      Object {
                        "marginLeft": 0,
                        "paddingLeft": 0,
                        "paddingTop": "0.25em",
                        "position": "relative",
                      }
                    }
                  >
                    <div
                      onClick={[Function]}
                      style={
                        Object {
                          "cursor": "pointer",
                          "display": "inline-block",
                          "paddingLeft": 0,
                          "paddingRight": "0.5em",
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "MozTransform": "rotateZ(90deg)",
                            "MozTransformOrigin": "45% 50%",
                            "MozTransition": "150ms",
                            "WebkitTransform": "rotateZ(90deg)",
                            "WebkitTransformOrigin": "45% 50%",
                            "WebkitTransition": "150ms",
                            "color": "#6FB3D2",
                            "fontSize": "0.75em",
                            "lineHeight": "1.1em",
                            "marginLeft": 0,
                            "position": "relative",
                            "transform": "rotateZ(90deg)",
                            "transformOrigin": "45% 50%",
                            "transition": "150ms",
                          }
                        }
                      >
                        ▶
                      </div>
                    </div>
                    <label
                      onClick={[Function]}
                      style={
                        Object {
                          "MozUserSelect": "inherit",
                          "WebkitUserSelect": "inherit",
                          "color": "#6FB3D2",
                          "cursor": "pointer",
                          "display": "inline-block",
                          "margin": 0,
                          "padding": 0,
                        }
                      }
                    >
                      <span>
                        state
                        :
                      </span>
                    </label>
                    <span
                      onClick={[Function]}
                      style={
                        Object {
                          "color": "#BEBEBE",
                          "cursor": "default",
                          "paddingLeft": "0.5em",
                        }
                      }
                    >
                      <span>
                        <span
                          style={
                            Object {
                              "marginLeft": "0.3em",
                              "marginRight": "0.3em",
                            }
                          }
                        >
                          {}
                        </span>
                         
                        1 key
                      </span>
                    </span>
                    <ul
                      style={
                        Object {
                          "display": "block",
                          "listStyle": "none",
                          "margin": 0,
                          "padding": 0,
                        }
                      }
                    >
                      <li
                        style={
                          Object {
                            "marginLeft": "0.875em",
                            "paddingLeft": 0,
                            "paddingTop": "0.25em",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                              "display": "inline-block",
                              "paddingLeft": 0,
                              "paddingRight": "0.5em",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "MozTransform": "rotateZ(0deg)",
                                "MozTransformOrigin": "45% 50%",
                                "MozTransition": "150ms",
                                "WebkitTransform": "rotateZ(0deg)",
                                "WebkitTransformOrigin": "45% 50%",
                                "WebkitTransition": "150ms",
                                "color": "#6FB3D2",
                                "fontSize": "0.75em",
                                "lineHeight": "1.1em",
                                "marginLeft": 0,
                                "position": "relative",
                                "transform": "rotateZ(0deg)",
                                "transformOrigin": "45% 50%",
                                "transition": "150ms",
                              }
                            }
                          >
                            ▶
                          </div>
                        </div>
                        <label
                          onClick={[Function]}
                          style={
                            Object {
                              "MozUserSelect": "inherit",
                              "WebkitUserSelect": "inherit",
                              "color": "#6FB3D2",
                              "cursor": "pointer",
                              "display": "inline-block",
                              "margin": 0,
                              "padding": 0,
                            }
                          }
                        >
                          <span>
                            overview
                            :
                          </span>
                        </label>
                        <span
                          onClick={[Function]}
                          style={
                            Object {
                              "color": "#A1C659",
                              "cursor": "default",
                              "paddingLeft": "0.5em",
                            }
                          }
                        >
                          <span>
                            <span
                              style={
                                Object {
                                  "marginLeft": "0.3em",
                                  "marginRight": "0.3em",
                                }
                              }
                            >
                              {}
                            </span>
                             
                            4 keys
                          </span>
                        </span>
                        <ul
                          style={
                            Object {
                              "display": "block",
                              "listStyle": "none",
                              "margin": 0,
                              "padding": 0,
                            }
                          }
                        />
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
